src/components/chip/chip.tsx

```diff
@@ -0,0 +1,289 @@
+import {
+    Component,
+    Element,
+    Event,
+    EventEmitter,
+    h,
+    Host,
+    Prop,
+} from '@stencil/core';
+import { Icon, Languages, Link } from '../../interface';
+import { getIconName } from '../icon/get-icon-props';
+import {
+    makeEnterClickable,
+    removeEnterClickable,
+} from '../../util/make-enter-clickable';
+import translate from '../../global/translations';
+import {
+    BACKSPACE,
+    BACKSPACE_KEY_CODE,
+    DELETE,
+    DELETE_KEY_CODE,
+} from '../../util/keycodes';
+
+/**
+ * Chips and buttons are both interactive elements in UI design,
+ * but they serve different purposes and are used in different contexts.
+ *
+ * :::warning
+ * Do not use the chip component carelessly, as an alternative for
+ * [`limel-button`](#/component/limel-button/) in the UI design!
+ *
+ * **Buttons:**
+ * Buttons are used to trigger actions. They are typically used to
+ * submit forms, open dialogs, initiate a process, or perform any action
+ * that changes the state of the application.
+ * Buttons' labels usually contain action words, in other words, the labels is
+ * a _verb in imperative mood_ such as "Submit" or "Delete".
+ * Buttons are placed in areas where it's clear they will initiate
+ * an action when clicked.
+ *
+ * **Chips:**
+ * Chips however are elements which may look like buttons, but they are
+ * representing choices, filters, or tags, in a small block
+ * or clearly bundled into a group. Chips are rarely used alone in the
+ * user interface.
+ * They are often used in a so called "chip-set", or placed together in
+ * a section of the UI, where the user can expect more than one chip to be present.
+ *
+ * For example, a chip may represent a filter in a filter bar, or a tag in a tag list,
+ * or an item in a shopping list.
+ * Clicking a chip can also trigger an action, for example toggling a filter ON or OFF,
+ * or opening a page with all posts tagged with the tag represented by the chip,
+ * or navigating to a page with more information about the item in the shopping list.
+ * :::
+ *
+ * @private
+ * @exampleComponent limel-example-chip-button
+ * @exampleComponent limel-example-chip-link
+ * @exampleComponent limel-example-chip-icon-colors
+ * @exampleComponent limel-example-chip-badge
+ * @exampleComponent limel-example-chip-filter
+ * @exampleComponent limel-example-chip-removable
+ * @exampleComponent limel-example-chip-aria-role
+ */
+@Component({
+    tag: 'limel-chip',
+    shadow: true,
+    styleUrl: 'chip.scss',
+})
+export class Chip {
```

Comment:
https://github.com/Lundalogik/lime-elements/pull/2707#discussion_r1457371745

This component should probably implement  the `Chip` interface (the existing one or a new one, but probably the existing one), to make sure the component and the interface are in sync.

Reply:
⚡

Since the `id` property is required in the `Chip` interface, but we removed it from here, I made a special solution that I think is good enough for now. The component is still private after all.

<img width="767" alt="image" src="https://github.com/Lundalogik/lime-elements/assets/154725/a78856e7-157f-4729-8c99-9ab9f479d46e">